components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/impl/ActionExecutorServiceImpl.java

```diff
@@ -312,18 +320,16 @@ private ActionExecutionStatus processActionResponse(Action action,
         } else if (actionInvocationResponse.isError() && actionInvocationResponse.getResponse() != null) {
             return processErrorResponse(action, (ActionInvocationErrorResponse) actionInvocationResponse.getResponse(),
                     eventContext, actionRequest, actionExecutionResponseProcessor);
-        } else {
-            logErrorResponse(action, actionInvocationResponse);
         }
-
-        return new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext);
+        logErrorResponse(action, actionInvocationResponse);
+        throw new ActionExecutionResponseProcessorException("Received an invalid or unexpected action status.");
```

Comment:
It's not a good practice to throw ActionExecutionResponseProcessorException here as that is intended by the respective SPI provided. 

You should rather throw an ActionExecutionRuntimeException, as this is an unexpected error that happens at runtime

Reply:
This is fixed with e01617689c045e164749dedaaf2f7f0603920ae2